Add Detekt in quality tools.

LoicDumasAQF opened the pull request:
I add the Kotlin code style tool Detekt.
Ktlint is problematic as it's impossible to override rules in ktlint.

I remove de check task dependency on ktlint task to replace it by detekt task.

jcayzac commented:
Do we need to keep both?

LoicDumasAQF commented:
Ktlint and detekt have different rule set. So ktlint will maybe return error not detected by detekt.  
So I didn't removed ktlint in case of someone want to use both.

But after more reflexion, it's better to remove Ktlint to avoid confusion, as some rules are in conflict. (wildcart usage, indentation, ...).
I will update the PR.

NemoOudeis commented:
sorry for the late reply, I forgot to "submit" the review... GitHub can be quite confusing at times... =(

LoicDumasAQF commented on quality/detekt/detekt-config.yml:
```diff
@@ -0,0 +1,383 @@
+autoCorrect: true
+failFast: false
+
```
I took the default [detekt-config.yml](https://github.com/arturbosch/detekt/blob/master/detekt-cli/src/main/resources/default-detekt-config.yml) found on detekt Github, so I didn't really checked all rules. 
But you're right, some rules should be activated, I will check it in detail and update it.

LoicDumasAQF commented on quality/README.md:
```diff
@@ -76,4 +79,14 @@ apply from: '../config/quality/errorprone/android.gradle'
+
+
+## Detekt
+Kotlin check style tool.
```
True, I changed it.

LoicDumasAQF commented:
I added a lot of rules, now only 15 are disabled. Thank you for noticing !
This configuration seems good to me, tried it on our project, and returned error where interesting. After, with more experience in Kotlin some of the rules should be enabled or disabled.
Anyway, with all these rules, Detekt is way better than Ktlint.

I also updated the `Readme.md`, tell me if the change log is good for you. With the "HEAD" version you added, I didn't knew how to add my change.

NemoOudeis commented:
Yeah I noticed that the links to the rule documentation in the README are broken, but found found them on the github pages site https://arturbosch.github.io/detekt/index.html
